test/realm_test.dart

```diff
@@ -552,5 +552,161 @@ Future<void> main([List<String>? args]) async {
       expect(() => realm.write(() => players[-1] = Person()), throws<RealmException>("Index out of range"));
       expect(() => realm.write(() => players[800] = Person()), throws<RealmException>());
     });
+
+    test('RealmList clear items from list', () {
+      var config = Configuration([Team.schema, Person.schema]);
+      var realm = Realm(config);
+
+      //Create a team
+      final team = Team()..name = "Team";
+      realm.write(() => realm.add(team));
+
+      //Add players to the team
+      final newPlayers = [
+        Person()..name = "Michael Schumacher",
+        Person()..name = "Sebastian Vettel",
+        Person()..name = "Kimi Räikkönen",
+      ];
+      realm.write(() {
+        team.players.addAll(newPlayers);
+      });
+
+      //Ensure teams and players are in realm
+      var teams = realm.all<Team>();
+      expect(teams.length, 1);
+
+      var players = teams[0].players;
+      expect(players, isNotNull);
+      expect(players.length, 3);
+
+      //Clear list of team players
+      realm.write(() => teams[0].players.clear());
+
+      //Ensure that players are not reffered by the team in realm
+      teams = realm.all<Team>();
+      expect(teams[0].players.length, 0);
+
+      //Ensure that players objects still exist in realm detached from the team
+      final allPlayers = realm.all<Person>();
+      expect(allPlayers.length, 3);
+    });
+
+    test('RealmList clear - same list related to two objects', () {
+      var config = Configuration([Team.schema, Person.schema]);
+      var realm = Realm(config);
+
+      //Create two teams
+      final teamOne = Team()..name = "TeamOne";
+      final teamTwo = Team()..name = "TeamTwo";
+      realm.write(() {
+        realm.add(teamOne);
+        realm.add(teamTwo);
+      });
+
+      //Create common players list for both teams
+      final newPlayers = [
+        Person()..name = "Michael Schumacher",
+        Person()..name = "Sebastian Vettel",
+        Person()..name = "Kimi Räikkönen",
+      ];
+      realm.write(() {
+        teamOne.players.addAll(newPlayers);
+        teamTwo.players.addAll(newPlayers);
+      });
+
+      //Ensure that teams and players exist in realm
+      var teams = realm.all<Team>();
+      expect(teams.length, 2);
+      expect(teams[0].players, isNotNull);
+      expect(teams[0].players.length, 3);
+      expect(teams[1].players, isNotNull);
+      expect(teams[1].players.length, 3);
+
+      //Clear first team's players only
+      realm.write(() => teams[0].players.clear());
+
+      //Reload teams from realm and ensure that second team is still related to players
```

Comment:
I suggest we do not requery the realm here.

Reply:
Done.